### ISSUE TITLE
Discussion: ExtendedPeoplePicker & Component Composition

### ISSUE THREAD

**Issue by Adjective-Object** [OfficeDev/office-ui-fabric-react#7872]
**Describe the feature that you would like added**

Right now extended people picker has renderprops that are rendered unconditionally.
This encourages boilerplate private functions [in order to avoid forced re-renders from lambdas](https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36?gi=adbcd081d75a)

See the extended people picker docs for [an example](https://github.com/OfficeDev/office-ui-fabric-react/blob/2ce6c5ac3ccf4e62d4866841317b4aa55b8fb541/packages/office-ui-fabric-react/src/components/ExtendedPicker/examples/ExtendedPeoplePicker.Basic.Example.tsx#L204):

```
      <ExtendedPeoplePicker
        ...
        onRenderFloatingPicker={this._onRenderFloatingPicker}
        onRenderSelectedItems={this._onRenderSelectedItems}
        ...
        />

...

  private _onRenderFloatingPicker(props: IBaseFloatingPickerProps<IPersonaProps>): JSX.Element {
    return <FloatingPeoplePicker {...props} />;
  }

  private _onRenderSelectedItems(props: IBaseSelectedItemsListProps<IExtendedPersonaProps>): JSX.Element {
    return <SelectedPeopleList {...props} />;
  }
```

[The React Docs](https://reactjs.org/docs/composition-vs-inheritance.html#containment) point to component containment / slotting as a way to solve the problem of aparent specifying specific descendants of a child component.

**What component or utility would this be added to**

ExtendedPeoplePicker

**Have you discussed this feature with our team, and if so, who**

opening this issue to open the discussion 😛 

**Additional context/screenshots**

**Comment by Adjective-Object** [comment]
ah I see, the floating pickers are mounting relative to a ref that the ExtendedPeoplePicker controls, so they need to pass down the ref into the render of the component.

Personally, I still think this leaves us with a messy API and a lot of boilerplate & I'd still like to try to tackle this somehow.

**Comment by Adjective-Object** [comment]
It's also curious why the rest of the props need to go from the parent, into the child, and then back up into the parent for render if the only thing we need is the component ref

**Comment by Adjective-Object** [comment]
Closed by #7877
